src/Runner.js

```diff
@@ -1,61 +1,57 @@
 import Namespace from './Namespace'
+import Resources from './Resources'
 import Resource from './Resource'
 import Root from './Root'
 
-export default class Duxy {
+export default class Runner {
   constructor (options = {}) {
-    this.options = options
+    this.dsl = {
+      resources: this.resources.bind(this),
+      resource: this.resource.bind(this),
+      namespace: this.namespace.bind(this),
+      get: this.get.bind(this),
+      post: this.post.bind(this),
+      put: this.put.bind(this),
+      patch: this.patch.bind(this),
+      del: this.del.bind(this),
+      map: this.map.bind(this),
+    }
 
-    this.root = new Root(this, options.http)
+    this.options = options
+    this.root = new Root(this.dsl, options.http)
     this.stack = [this.root]
+  }
 
-    // NOTE(vesln): DSL methods
-    this.resource = this.resource.bind(this)
-    this.namespace = this.namespace.bind(this)
-    this.get = this.get.bind(this)
-    this.post = this.post.bind(this)
-    this.put = this.put.bind(this)
-    this.patch = this.patch.bind(this)
-    this.del = this.del.bind(this)
-    this.map = this.map.bind(this)
+  resources (name, options, fn) {
+    this.addRunnable(new Resources(this.current, name, options, fn))
   }
 
   resource (name, options, fn) {
-    const resource = new Resource(this.current, name, options, fn)
-    this.current.runnables.push(resource)
-
-    this.stack.push(resource)
-    resource.run()
-    this.stack.pop()
+    this.addRunnable(new Resource(this.current, name, options, fn))
   }
 
   namespace (name, options, fn) {
-    const namespace = new Namespace(this.current, name, options, fn)
-    this.current.runnables.push(namespace)
-
-    this.stack.push(namespace)
-    namespace.run()
-    this.stack.pop()
+    this.addRunnable(new Namespace(this.current, name, options, fn))
   }
 
-  get (name) {
-    this.member(name, 'GET')
+  get (name, options) {
```

Comment:
```js
get('edit', { path: '{id}/edit' })
```

I'm leaning towars the above because:

- it's more rails like compatible
- more explicit
- more flexible fot future features